authorize.go

```diff
@@ -33,14 +33,11 @@ func abs(d time.Duration) time.Duration {
 	return d
 }
 
-// authorize validates the hawk authorization header on a request
-// and returns the userid and a boolean indicating authorization status
-func (a *autographer) authorize(r *http.Request, body []byte) (userid string, authorize bool, err error) {
-	var (
-		auth *hawk.Auth
-	)
+// authorizeHeader validates the hawk authorization header on a request
+// and returns the hawk auth struct, the userid and a boolean indicating authorization status
```

Comment:
nit: docstring is out of date

nit: I forgot to mention this earlier, but since it looks like `authorize` is always false when an error is returned and true when an error isn't returned we can drop the authorize param and just check `error == nil` to get authorize/authorized